### PR TITLE
Onboarding: remove /new (Gutenboarding & AnchorFM) flows

### PR DESCRIPTION
**Pull request by fushar**
## Proposed Changes

This PR cleans up old code around Gutenboarding and AnchorFM flows.

## Testing Instructions

Smoke test the following flows:

* General onboarding (`/setup`)
* VideoPress (`/setup/videopress`), verify we can select both videomaker and the white variation (and actually applied to the site)
* Sensei (`/setup/sensei`), verify that we can finish the flow
* DIFM (`/start/do-it-for-me`), verify that the design picker still loads fine
   - I was not able to verify that the selected design is correct, even in production. Because the created site seems to use the default theme (TT4) no matter what theme I selected in the design picker step... any input is welcome. Maybe we only activate the design manually based on the content request.
   - I've at least verified that the `calypso_signup_select_design` event is fired correctly.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/design-picker/src/types.ts [hunk 1]
```diff
@@ -114,20 +101,9 @@ export interface Design {
 	preview_data?: PreviewData;
 	screenshot?: string;
 
-	/** @deprecated used for Gutenboarding (/new flow) */
+	/** @deprecated TODO: replace both with just stylesheet */
```
Comment (fushar) [inline packages/design-picker/src/types.ts:104]: I will resolve this in a follow-up PR (combining recipe, stylesheet, and theme) when I have another 20% time 🥲

File: client/signup/steps/design-picker/index.jsx [hunk 2]
```diff
@@ -51,12 +42,10 @@ export default function DesignPickerStep( props ) {
 		siteHasFeature( state, siteId, WPCOM_FEATURES_PREMIUM_THEMES )
 	);
 
-	const userLoggedIn = useSelector( ( state ) => isUserLoggedIn( state ) );
-
 	const dispatch = useDispatch();
 	const translate = useTranslate();
 
-	const [ selectedDesign, setSelectedDesign ] = useState( null );
+	const [ selectedDesign ] = useState( null );
```
Comment (jsnajdr) [inline client/signup/steps/design-picker/index.jsx]: Could the `selectedDesign` variable be eliminated completely? It's now always `null`.

Comment (fushar) [reply client/signup/steps/design-picker/index.jsx]: Nice catch, done, thanks!

**Review by arthur791004 — APPROVED** [review]
Tested and it looks good. Nice job 🎉

✅ General onboarding (/setup)
✅ VideoPress (/setup/videopress)
✅ Sensei (/setup/sensei)
✅ DIFM (/start/do-it-for-me)

**Comment by arthur791004** [comment]
Added the reference p7DVsv-ha2-p2, Heads-up: Sunsetting Anchor.fm

**Review by tyxla — COMMENTED** [review]
Love to see all the cleanup here ❤️

**Comment by fushar** [comment]
Thanks all. Will be shipping this soon 🤞

**Pull request merged**